sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -264,19 +264,43 @@ export class ServiceBusClient {
-    createDeadLetterReceiver(queueName: string, receiveMode: "receiveAndDelete"): Receiver<ReceivedMessage>;
-    createDeadLetterReceiver(topicName: string, subscriptionName: string, receiveMode: "peekLock"): Receiver<ReceivedMessageWithLock>;
-    createDeadLetterReceiver(topicName: string, subscriptionName: string, receiveMode: "receiveAndDelete"): Receiver<ReceivedMessage>;
-    createReceiver(queueName: string, receiveMode: "peekLock"): Receiver<ReceivedMessageWithLock>;
-    createReceiver(queueName: string, receiveMode: "receiveAndDelete"): Receiver<ReceivedMessage>;
-    createReceiver(topicName: string, subscriptionName: string, receiveMode: "peekLock"): Receiver<ReceivedMessageWithLock>;
-    createReceiver(topicName: string, subscriptionName: string, receiveMode: "receiveAndDelete"): Receiver<ReceivedMessage>;
+    createDeadLetterReceiver(queueName: string, options?: {
+        receiveMode?: "peekLock";
+    }): Receiver<ReceivedMessageWithLock>;
+    createDeadLetterReceiver(queueName: string, options?: {
+        receiveMode?: "receiveAndDelete";
+    }): Receiver<ReceivedMessage>;
+    createDeadLetterReceiver(topicName: string, subscriptionName: string, options?: {
+        receiveMode?: "peekLock";
+    }): Receiver<ReceivedMessageWithLock>;
+    createDeadLetterReceiver(topicName: string, subscriptionName: string, options?: {
+        receiveMode?: "receiveAndDelete";
+    }): Receiver<ReceivedMessage>;
+    createReceiver(queueName: string, options?: {
+        receiveMode?: "peekLock";
+    }): Receiver<ReceivedMessageWithLock>;
+    createReceiver(queueName: string, options?: {
+        receiveMode?: "receiveAndDelete";
+    }): Receiver<ReceivedMessage>;
+    createReceiver(topicName: string, subscriptionName: string, options?: {
+        receiveMode?: "peekLock";
+    }): Receiver<ReceivedMessageWithLock>;
+    createReceiver(topicName: string, subscriptionName: string, options?: {
+        receiveMode?: "receiveAndDelete";
+    }): Receiver<ReceivedMessage>;
     createSender(queueOrTopicName: string): Sender;
-    createSessionReceiver(queueName: string, receiveMode: "peekLock", options?: CreateSessionReceiverOptions): Promise<SessionReceiver<ReceivedMessageWithLock>>;
-    createSessionReceiver(queueName: string, receiveMode: "receiveAndDelete", options?: CreateSessionReceiverOptions): Promise<SessionReceiver<ReceivedMessage>>;
-    createSessionReceiver(topicName: string, subscriptionName: string, receiveMode: "peekLock", options?: CreateSessionReceiverOptions): Promise<SessionReceiver<ReceivedMessageWithLock>>;
-    createSessionReceiver(topicName: string, subscriptionName: string, receiveMode: "receiveAndDelete", options?: CreateSessionReceiverOptions): Promise<SessionReceiver<ReceivedMessage>>;
+    createSessionReceiver(queueName: string, options?: CreateSessionReceiverOptions & {
+        receiveMode?: "peekLock";
```

Comment:
We can have receiveMode pulled into `CreateSessionReceiverOptions`? Are we not doing that so that we can have distinct output types ReceiveMessage vs ReceiveMessageWithLock?

Reply:
No.
Yes.